### ISSUE TITLE
runtime issues

### ISSUE THREAD

**Issue by neok-m4700** [JuliaInterop/Cxx.jl#351]
What is the highest LLVM version supported ? 

Using LLVM@4.0.1 with julia@0.6.0 yields the following:

```bash
$ julia -e 'Pkg.add("Cxx"); Pkg.test("Cxx")'
INFO: Cloning cache of Cxx from https://github.com/Keno/Cxx.jl.git
INFO: Installing Cxx v0.2.0
INFO: Building Cxx
INFO: Building julia source build
INFO: Package database updated
INFO: Testing Cxx
julia_: $LLVM/include/llvm/ADT/StringSet.h:33: std::pair<typename llvm::StringMap<char, AllocatorTy>::iterator, bool> llvm::StringSet<AllocatorTy>::insert(llvm::StringRef) [with AllocatorTy = llvm::MallocAllocator; typename llvm::StringMap<char, AllocatorTy>::iterator = llvm::StringMapIterator<char>]: 
Assertion `!Key.empty()' failed.

signal (6): Aborted
while loading $PREFIX/share/julia/site/v0.6/Cxx/test/runtests.jl, in expression starting on line 69
__libc_signal_restore_set at /build/glibc-mXZSwJ/glibc-2.24/signal/../sysdeps/unix/sysv/linux/nptl-signals.h:79 [inlined]
raise at /build/glibc-mXZSwJ/glibc-2.24/signal/../sysdeps/unix/sysv/linux/raise.c:55
abort at /build/glibc-mXZSwJ/glibc-2.24/stdlib/abort.c:89
__assert_fail_base at /build/glibc-mXZSwJ/glibc-2.24/assert/assert.c:92
__assert_fail at /build/glibc-mXZSwJ/glibc-2.24/assert/assert.c:101
_ZN4llvm12GlobalObject10setSectionENS_9StringRefE at $PREFIX/bin/../lib/libLLVM-4.0.so (unknown line)
CloneFunctionBody at $JULIA/src/ccall.cpp:880 [inlined]
ResolveLazyFunctions at $JULIA/src/ccall.cpp:896 [inlined]
CloneFunction at $JULIA/src/ccall.cpp:886 [inlined]
emit_llvmcall at $JULIA/src/ccall.cpp:1140
emit_intrinsic at $JULIA/src/intrinsics.cpp:757
emit_call at $JULIA/src/codegen.cpp:3434
emit_expr at $JULIA/src/codegen.cpp:4139
emit_function at $JULIA/src/codegen.cpp:6095
jl_compile_linfo at $JULIA/src/codegen.cpp:1256
jl_compile_for_dispatch at $JULIA/src/gf.c:1675
jl_compile_method_internal at $JULIA/src/julia_internal.h:307 [inlined]
jl_call_method_internal at $JULIA/src/julia_internal.h:354 [inlined]
jl_apply_generic at $JULIA/src/gf.c:1933
do_call at $JULIA/src/interpreter.c:75
eval at $JULIA/src/interpreter.c:242
eval_body at $JULIA/src/interpreter.c:543
jl_interpret_toplevel_thunk at $JULIA/src/interpreter.c:692
jl_toplevel_eval_flex at $JULIA/src/toplevel.c:592
jl_toplevel_eval_in at $JULIA/src/builtins.c:496
RunGlobalConstructors at $PREFIX/share/julia/site/v0.6/Cxx/src/initialization.jl:69
ParseToEndOfFile at $PREFIX/share/julia/site/v0.6/Cxx/src/initialization.jl:115
cxxparse at $PREFIX/share/julia/site/v0.6/Cxx/src/initialization.jl:134
unknown function (ip: 0x7f5088303a54)
jl_call_fptr_internal at $JULIA/src/julia_internal.h:339 [inlined]
jl_call_method_internal at $JULIA/src/julia_internal.h:358 [inlined]
jl_apply_generic at $JULIA/src/gf.c:1933
macro expansion at $PREFIX/share/julia/site/v0.6/Cxx/src/cxxstr.jl:685 [inlined]
macro expansion at $PREFIX/share/julia/site/v0.6/Cxx/test/runtests.jl:8 [inlined]
macro expansion at ./test.jl:860 [inlined]
anonymous at ./<missing> (unknown line)
jl_call_fptr_internal at $JULIA/src/julia_internal.h:339 [inlined]
jl_call_method_internal at $JULIA/src/julia_internal.h:358 [inlined]
jl_toplevel_eval_flex at $JULIA/src/toplevel.c:589
jl_parse_eval_all at $JULIA/src/ast.c:873
jl_load at $JULIA/src/toplevel.c:616
include_from_node1 at ./loading.jl:569
unknown function (ip: 0x7f50ba43237b)
jl_call_fptr_internal at $JULIA/src/julia_internal.h:339 [inlined]
jl_call_method_internal at $JULIA/src/julia_internal.h:358 [inlined]
jl_apply_generic at $JULIA/src/gf.c:1933
include at ./sysimg.jl:14
unknown function (ip: 0x7f50ba2d203b)
jl_call_fptr_internal at $JULIA/src/julia_internal.h:339 [inlined]
jl_call_method_internal at $JULIA/src/julia_internal.h:358 [inlined]
jl_apply_generic at $JULIA/src/gf.c:1933
process_options at ./client.jl:305
_start at ./client.jl:371
unknown function (ip: 0x7f50ba43e228)
jl_call_fptr_internal at $JULIA/src/julia_internal.h:339 [inlined]
jl_call_method_internal at $JULIA/src/julia_internal.h:358 [inlined]
jl_apply_generic at $JULIA/src/gf.c:1933
jl_apply at $JULIA/ui/../src/julia.h:1424 [inlined]
true_main at $JULIA/ui/repl.c:127
main at $JULIA/ui/repl.c:264
__libc_start_main at /build/glibc-mXZSwJ/glibc-2.24/csu/../csu/libc-start.c:291
unknown function (ip: 0x401729)
Allocations: 2727107 (Pool: 2725756; Big: 1351); GC: 5
=================================[ ERROR: Cxx ]=================================

failed process: Process(`$PREFIX/bin/julia -Cnative -J$PREFIX/lib/julia/sys.so --compile=yes --depwarn=yes --check-bounds=yes --code-coverage=none --color=no --compilecache=yes $PREFIX/share/julia/site/v0.6/Cxx/test/runtests.jl`, ProcessSignaled(6)) [0]

================================================================================
ERROR: Cxx had test errors
```
Ubuntu 17.04

**Comment by neok-m4700** [comment]
Even simpler test, exact issue ...

```bash
$ julia -e 'using Cxx; cxx""" #include<iostream> """'
```

**Comment by r9y9** [comment]
It seems LLVM 4.0 support is not fully done yet. This doesn't happen with LLVM 3.9.1 on Julia v0.6.2.

**Comment by ararslan** [comment]
I think in general you'll have the easiest time using the LLVM that ships with Julia, since we need some of the Julia patches.

**Comment by r9y9** [comment]
Looks like https://github.com/llvm-mirror/llvm/commit/b863b6b817c34c5714c3bdf08e55980e8e245083 fixes the issue. However, there's another assertion failure while running `Pkg.test("Cxx")`. I think I'll probably have to wait for LLVM experts to fix issues for LLVM 4.0.

**Comment by Gnimuc** [comment]
closed since the issue is obsoleted.
